packages/iota/src/binary/milestoneOptions/milestoneOptions.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+import type { ReadStream, WriteStream } from "@iota/util.js";
+import type { ITypeBase } from "../../models/ITypeBase";
+import { IPoWMilestoneOption, POW_MILESTONE_OPTION_TYPE } from "../../models/milestoneOptions/IPoWMilestoneOption";
+import { IReceiptMilestoneOption, RECEIPT_MILESTONE_OPTION_TYPE } from "../../models/milestoneOptions/IReceiptMilestoneOption";
+import type { MilestoneOptionTypes } from "../../models/milestoneOptions/milestoneOptionTypes";
+import {
+    deserializePoWMilestoneOption,
+    MIN_POW_MILESTONE_OPTION_LENGTH,
+    serializePoWMilestoneOption
+} from "../milestoneOptions/powMilestoneOption";
+import {
+    deserializeReceiptMilestoneOption,
+    MIN_RECEIPT_MILESTONE_OPTION_LENGTH,
+    serializeReceiptMilestoneOption
+} from "../milestoneOptions/receiptMilestoneOption";
+
+/**
+ * The minimum length of a milestone option binary representation.
+ */
+ export const MIN_MILESTONE_OPTION_LENGTH: number = Math.min(
+    MIN_RECEIPT_MILESTONE_OPTION_LENGTH,
+    MIN_POW_MILESTONE_OPTION_LENGTH
+);
+
+
+/**
+ * Deserialize the milestone options from binary.
+ * @param readStream The stream to read the data from.
+ * @returns The deserialized object.
+ */
+ export function deserializeMilestoneOptions(readStream: ReadStream): MilestoneOptionTypes[] {
+    const numMilestoneOptions = readStream.readUInt8("milestoneOptions.numMilestoneOptions");
```

Comment:
Should we have the same naming here as in the serialize side on line 50 ? And maybe name the constant like that also ? 
i.e. `const optionsCount = readStream.readUInt8("milestoneOptions.optionsCount");`